Source/Plugins/Tilemaps/Core/Tilemaps/TilemapFloodFill.cs

```diff
@@ -0,0 +1,213 @@
+		/// <param name="tilemap"></param>
+		/// <param name="pos"></param>
+		/// <param name="preview">If true, the algorithm will cancel when taking too long for an interactive preview.</param>
+		/// <param name="floodFillArea"></param>
+		/// <param name="floodFillOrigin"></param>
+		/// <returns>True, if the algorithm completed. False, if it was canceled.</returns>
+		public bool GetFillArea(Tilemap tilemap, Point2 pos, bool preview, Grid<bool> floodFillArea, ref Point2 floodFillOrigin)
+		{
+			Grid<Tile> tiles = tilemap.BeginUpdateTiles();
+			Point2 fillTopLeft;
+			Point2 fillSize;
+			bool success = FillArea(ref this.activeFillBuffer, tiles, pos, preview ? (128 * 128) : 0, out fillTopLeft, out fillSize);
+			tilemap.EndUpdateTiles(0, 0, 0, 0);
+
+			// Find the filled areas boundaries and copy it to the active area
+			if (success)
+			{
+				floodFillOrigin = fillTopLeft;
+				floodFillArea.ResizeClear(fillSize.X, fillSize.Y);
+				this.activeFillBuffer.CopyTo(floodFillArea, 0, 0, -1, -1, floodFillOrigin.X, floodFillOrigin.Y);
+			}
+
+			return success;
+		}
+
+		/// <summary>
+		/// Performs a flood fill operation originating from the specified position.
+		/// <see cref="Tile"/> equality is checked in the <see cref="FillAreaTilesEqual"/> method.
+		/// </summary>
+		/// <param name="fillBuffer">A buffer that will be filled with the result of the flood fill operation.</param>
+		/// <param name="tiles"></param>
+		/// <param name="pos"></param>
+		/// <param name="maxTileCount">The maximum number of tiles to fill. If the filled tile count exceeds this number, the algorithm is canceled.</param>
+		/// <param name="fillAreaSize"></param>
+		/// <param name="fillAreaTopLeft"></param>
+		/// <returns>True when successful. False when aborted.</returns>
+		public bool FillArea(ref Grid<bool> fillBuffer, Grid<Tile> tiles, Point2 pos, int maxTileCount, out Point2 fillAreaTopLeft, out Point2 fillAreaSize)
+		{
+			// ## Note: ##
+			// This flood fill algorithm is a modified version of "A More Efficient Flood Fill" by Adam Milazzo.
+			// All credit for the original idea and sample implementation goes to him. Last seen on the web here:
+			// http://adammil.net/blog/v126_A_More_Efficient_Flood_Fill.html
+			// ###########
+
+			// Initialize fill buffer
+			if (fillBuffer == null)
+				fillBuffer = new Grid<bool>(tiles.Width, tiles.Height);
+			else if (fillBuffer.Width != tiles.Width || fillBuffer.Height != tiles.Height)
+				fillBuffer.ResizeClear(tiles.Width, tiles.Height);
+			else
+				fillBuffer.Clear();
+
+			// Get the base tile for comparison
+			Tile baseTile = tiles[pos.X, pos.Y];
+
+			// Find the topleft-most tile to start with
+			fillAreaTopLeft = FillAreaFindTopLeft(fillBuffer, tiles, pos, baseTile);
+			fillAreaSize = new Point2(1 + pos.X - fillAreaTopLeft.X, 1 + pos.Y - fillAreaTopLeft.Y);
+			pos = fillAreaTopLeft;
+
+			// Run the main part of the algorithm
+			if (maxTileCount <= 0) maxTileCount = int.MaxValue;
```

Comment:
Given that this we are already leaving off the `{}` on single lined if blocks this should either be moved down and tabbed up or a space should follow so the assignment doesn't seem related.